src/gwt/src/org/rstudio/studio/client/common/filetypes/ScriptFileType.java

```diff
@@ -78,6 +79,25 @@ public String getScriptInterpreter()
          result.add(commands.sourceActiveDocument());
          result.add(commands.sourceActiveDocumentWithEcho());
       }
+
+      if (canExecuteCode())
```

Comment:
Yes that is the intent, and agree, it is ugly. I originally did it directly in TextFileType, but then convinced myself (barely) that maybe it should be something that was hidden in ScriptFileType. But TextFileType already has a bunch of file-type specific logic, so I'll move it back so it's easier to maintain.